File: domain/src/main/java/com/everis/domain/repository/UserRepository.java
package com.everis.domain.repository;

import com.everis.domain.model.User;

import rx.Observable;

/**
 * Created by everis on 25/04/18.
 */

public interface UserRepository {

    Observable<User> createUser(String email, String password);
    Observable<User> signInUser(String email, String password);
    Observable<User> checkUserLoggedIn();
    Observable<User> getUserState();
    Observable<Void> logOut();
    Observable<Void> deleteUser();

}

File: domain/src/main/java/com/everis/domain/repository/UserRepository.kt
package com.everis.domain.repository

import com.everis.domain.model.User

import rx.Observable

interface UserRepository {

    fun createUser(email: String, password: String): Observable<User>
    fun signInUser(email: String, password: String): Observable<User>
    fun checkUserLoggedIn(): Observable<User>
    fun logOut(): Observable<Void>
    fun deleteUser(): Observable<Void>
    fun getUserState(): Observable<User>

}
